scripts/zkai_attention_kv_stwo_native_two_head_bounded_softmax_table_proof_input.py

```
@@ -0,0 +1,782 @@
+        require_input_step(step, label=f"input_steps[{index}]")
+        for index, step in enumerate(require_sequence(input_steps, label="input_steps"))
+    ]
+    rows: list[dict[str, Any]] = []
+    outputs: list[list[int]] = []
+    local_step_counts = [0 for _ in range(HEAD_COUNT)]
+    for global_step_index, step in enumerate(steps):
+        head_index = step["head_index"]
+        step_index = local_step_counts[head_index]
+        local_step_counts[head_index] += 1
+        if step_index >= SEQUENCE_LENGTH:
+            raise AttentionKvTwoHeadBoundedSoftmaxTableInputError(f"input_steps[{global_step_index}] exceeds per-head sequence length")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)